llvm/test/CodeGen/NVPTX/kernel-param-align.ll

```diff
@@ -0,0 +1,36 @@
+; RUN: llc < %s -march=nvptx64 -mcpu=sm_72 2>&1 | FileCheck %s
+; RUN: %if ptxas %{ llc < %s -march=nvptx64 -mcpu=sm_72 | %ptxas-verify %}
+
+%struct.Large = type { [16 x double] }
+
+; CHECK-LABEL: func_align
+; CHECK: .param .u64 .ptr .global .align 16 func_align_param_0,
+; CHECK: .param .u64 .ptr .global func_align_param_1,
+; CHECK: .param .u32 func_align_param_2
+define void @func_align(ptr nocapture readonly align 16 %input, ptr addrspace(3) nocapture %out, i32 %n) {
+entry:
+  %0 = addrspacecast ptr %out to ptr addrspace(1)
+  %1 = addrspacecast ptr %input to ptr addrspace(1)
+  %getElem = getelementptr inbounds %struct.Large, ptr addrspace(1) %1, i64 0, i32 0, i64 5
+  %tmp2 = load i32, ptr addrspace(1) %getElem, align 8
+  store i32 %tmp2, ptr addrspace(1) %0, align 4
+  ret void
+}
+
+; CHECK-LABEL: func
+; CHECK: .param .ptr .global .u64 func_param_0,
```

Comment:
Did you actually run the tests? Did they succeed?

I'm somewhat puzzled that we end up with two different ways to generate parameter directives:

- CHECK: .param .u64 .ptr .global .align 16 func_align_param_0,
- CHECK: .param .ptr .global .u64 func_param_0,


How do we end up with the first kernel having `.u64` as the first directive, while the second kernel has it just before the identifier? AFAICT, `.u64` should always be emitted before the `.ptr`.

Reply:
Updated the test case and ran ptxas on the test case. Thanks.